Assets/BossRoom/Scripts/Shared/OnSpawnBehaviourEnabler.cs

```diff
@@ -0,0 +1,69 @@
+using System;
+using System.Collections.Generic;
+using Unity.Netcode;
+using UnityEngine;
+
+namespace Unity.Multiplayer.Samples.Utilities
+{
+    public interface IClientOnlyMonoBehaviour
+    {
+        public void SetEnabled(bool enable);
+    }
+
+    public interface IServerOnlyMonoBehaviour
+    {
+        public void SetEnabled(bool enable);
+    }
+
+
+    /// <summary>
+    /// Disables a list of MonoBehaviours on Awake, then only enables them when spawning based on whether this game instance
+    /// is a client or a server (or both in case of a client-hosted session).
+    /// </summary>
+    public class OnSpawnBehaviourEnabler : NetworkBehaviour
+    {
+        void Awake()
+        {
+            // Disable everything here to prevent those MonoBehaviours to be updated before this one is spawned.
+            DisableAll();
+        }
+
+        public override void OnNetworkSpawn()
+        {
+            if (IsClient)
+            {
+                foreach (var behaviour in gameObject.GetComponents<IClientOnlyMonoBehaviour>())
```

Comment:
OnNetworkSpawn would be called with network pooling correct? This has the potential to be a very costly spawn, GetComponent isn't performant

Reply:
especially if there's multiple multiple spawns with pooling

Reply:
Oh you're right, that would be super costly. Maybe I should cache all of this in its Awake method? Another solution could be to expose a list to the editor, but that would require us to manually enter all netbehaviours, but that would be even more work than what this tool is trying to simplify

Reply:
yeah caching would be good